helmctlr/remote_chart.go

```diff
@@ -0,0 +1,96 @@
+package helmctlr
+
+import (
+	"strings"
+
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"errors"
+
+	"encoding/base64"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/helm/pkg/downloader"
+	"k8s.io/helm/pkg/getter"
+	"k8s.io/helm/pkg/helm/environment"
+	"k8s.io/helm/pkg/helm/helmpath"
+)
+
+// getChartRef reads the chart entry from CR spec and returns chart entry or empty string.
+func getChartRef(r *unstructured.Unstructured) string {
+	if spec, ok := r.Object["spec"].(map[string]interface{}); ok {
+		if chart, ok := spec["chart"].(string); ok {
+			return chart
+		}
+	}
+
+	return ""
+}
+
+// Gets chart name and chart version from the passed chartRef, downloads the chart from repo,
```

Comment:
this should start with getRemoteChart to match godoc format.

Reply:
added it!